skimage/feature/blob.py

```diff
@@ -193,19 +216,24 @@ def blob_dog(image, min_sigma=1, max_sigma=50, sigma_ratio=1.6, threshold=2.0,
     # multiplying with standard deviation provides scale invariance
     dog_images = [(gaussian_images[i] - gaussian_images[i + 1])
                   * sigma_list[i] for i in range(k)]
-    image_cube = np.dstack(dog_images)
+
+    # Replace by image_cube = np.stack(hessian_images, axis=-1)
+    # When we upgrade minimal requirements to NumPy 1.10
+    sl = (slice(None),) * image.ndim + (np.newaxis,)
+    arrays = [np.asanyarray(arr) for arr in dog_images]
```

Comment:
LOL I hadn't seen this code before... `dog_images` definitely sounds like something other than what it is. =)